core/src/main/java/tech/pegasys/eth2signer/core/service/operations/PublicKeys.java

```diff
@@ -12,19 +12,28 @@
  */
 package tech.pegasys.eth2signer.core.service.operations;
 
+import static tech.pegasys.eth2signer.core.signing.KeyType.BLS;
+
 import tech.pegasys.eth2signer.core.signing.ArtifactSignerProvider;
+import tech.pegasys.eth2signer.core.signing.KeyType;
 
 import java.util.ArrayList;
 import java.util.List;
 
 public class PublicKeys {
-  private final ArtifactSignerProvider signerProvider;
+  private final ArtifactSignerProvider blsSignerProvider;
```

Comment:
Wondering if this is the right split (not sure) - i.e. should we maintain 2 different sets of keys everywhere - and build required handlers around each set.
I.e. this solution has a single handler, which then asks a single-keyhandler get me "bls" or "secp" keys - do we want to have 2 handlers, which are initialised with "SECP" and SecpSIgners, and another with "BLS" and BlsSigners?

Reply:
It was easier to do the split so that the rpc service and the rest api handler didn't need to have any knowledge about the signer provider split.